include/f_core/os/tenants/c_datalogger_tenant.h

```diff
@@ -0,0 +1,35 @@
+#ifndef C_DATALOGGER_TENANT_H
+#define C_DATALOGGER_TENANT_H
+
+#include <f_core/messaging/c_message_port.h>
+#include <f_core/os/c_tenant.h>
+#include <f_core/os/c_datalogger.h>
+#include <zephyr/logging/log.h>
+
+template <typename T>
+class CDataLoggerTenant : public CTenant {
+public:
+    CDataLoggerTenant(const char *name, const char *filename, LogMode mode, std::size_t num_packets, CMessagePort<T> &messagePort)
+        : CTenant(name), messagePort(messagePort), dataLogger(filename, mode, num_packets), filename(filename) {}
+
+    ~CDataLoggerTenant() override {
+        Cleanup();
+    }
+
+    void Run() override {
+        if (T message{}; messagePort.Receive(message, K_FOREVER) == 0) {
+            dataLogger.write(message);
+        }
+    }
+
+    void Cleanup() {
```

Comment:
should cleanup be a method for all tenants? just default do nothing and after stopRtos call cleanup on all of them?

Reply:
Yes in the future. I just didn't want to go and change every tenant right now